java-checks/src/main/java/org/sonar/java/checks/ThrowCheckedExceptionCheck.java

```diff
@@ -41,18 +46,53 @@
 @SqaleConstantRemediation("1h")
 public class ThrowCheckedExceptionCheck extends SubscriptionBaseVisitor {
 
+  private Stack<MethodTree> methods = new Stack<>();
+
   @Override
   public List<Tree.Kind> nodesToVisit() {
-    return ImmutableList.of(Tree.Kind.THROW_STATEMENT);
+    return ImmutableList.of(Tree.Kind.METHOD, Tree.Kind.THROW_STATEMENT);
   }
 
   @Override
   public void visitNode(Tree tree) {
-    ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
-    Type symbolType = throwStatementTree.expression().symbolType();
-    if (symbolType.isSubtypeOf("java.lang.Exception") && !symbolType.isSubtypeOf("java.lang.RuntimeException")) {
-      reportIssue(throwStatementTree.expression(), "Remove the usage of the checked exception '" + symbolType.name() + "'.");
+    if (tree.is(Tree.Kind.METHOD)) {
+      methods.push((MethodTree) tree);
+    } else {
+      ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
+      Type symbolType = throwStatementTree.expression().symbolType();
+      if (symbolType.isSubtypeOf("java.lang.Exception") && !symbolType.isSubtypeOf("java.lang.RuntimeException") && !isFromMethodOverride(symbolType, methods)) {
+        reportIssue(throwStatementTree.expression(), "Remove the usage of the checked exception '" + symbolType.name() + "'.");
+      }
+    }
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (tree.is(Tree.Kind.METHOD)) {
+      methods.pop();
+    }
+  }
+
+  private static boolean isFromMethodOverride(Type exceptionType, Stack<MethodTree> methods) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1149)